### PR TITLE
refactor: use go-colorful v1.4.0 for wide-gamut color transforms

### PR DESCRIPTION
**Pull request by bennypowers**
## Summary
- Update go-colorful from v1.3.0 to v1.4.0
- Replace hand-rolled CSS Color 4 transfer functions, primaries-to-XYZ matrices, and Bradford chromatic adaptation with native library calls (`colorful.DisplayP3()`, `colorful.A98Rgb()`, `colorful.ProPhotoRgb()`, `colorful.Rec2020()`, `colorful.XyzD50()`)
- Net -120 lines with no behavior change

Ref: lucasb-eyer/go-colorful#81

## Test plan
- [x] All existing Android formatter tests pass (including wide-gamut and low-value edge cases)
- [x] Full test suite passes
- [x] Lint clean

## Summary by CodeRabbit

* **Refactor**
  * Simplified and consolidated color space conversions used for Android formatting, replacing custom conversion math with standardized library conversions.

* **Tests**
  * Strengthened Android formatter tests to assert exact hex outputs for wide-gamut and edge-case color spaces, improving accuracy and coverage.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.25.5
 
 require (
 	github.com/bmatcuk/doublestar/v4 v4.9.2
-	github.com/lucasb-eyer/go-colorful v1.3.0
+	github.com/lucasb-eyer/go-colorful v1.4.0
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply go.mod]: and yet, ci is green.
